studiopress/genesis-custom-blocks
# Add a File field, like for .pdf uploads

# Changes
* Add a File field that allows non-image files:

<img width="823" alt="ioabsdifadsf" src="https://user-images.githubusercontent.com/4063887/125011796-bbdef280-e02e-11eb-9c59-378341d936c0.png">

* There's already an Image field, but that doesn't allow some files like `.pdf` files

Fixes [GF-3071](https://wpengine.atlassian.net/browse/GF-3071)

# Testing instructions
1. `cd /path/to/wp-content/plugins/genesis-custom-blocks/`

2. `git fetch`

3. `git checkout add/file-field`

4. `composer install && npm i && npm run build`

5. `/wp-admin` > Custom Blocks > Add New

6. Add a field with the type 'File': 
<img width="934" alt="file-here" src="https://user-images.githubusercontent.com/4063887/125011327-dd8baa00-e02d-11eb-8b1e-66031aae6bc7.png">

7. Click Publish

8. `/wp-admin` > Posts > Add New

9. Enter the name of the block you just created: 
<img width="850" alt="thebisdf" src="https://user-images.githubusercontent.com/4063887/125011464-23487280-e02e-11eb-8da4-b9c346db666e.png">

10. In the block, click 'Upload' and upload any `.pdf` file

11. Expected: The file name displays: 
<img width="823" alt="ioabsdifadsf" src="https://user-images.githubusercontent.com/4063887/125011659-7de1ce80-e02e-11eb-9f20-212210fa2508.png">

12. This won't preview the actual file in the Edit component, as the file could be a `.pdf`, a `.zip`, or any number of files
